src/main/java/com/amplitude/api/AmplitudeClient.java

```diff
@@ -1618,7 +1621,6 @@ public void onFinished() {
                     }, serverZone);
                 }
                 startNewSessionIfNeeded(timestamp);
```

Comment:
Thanks @falconandy, I see what you mean about the logic being different. Event before your changes the call to `startNewSessionIfNeeded(timestamp)` should be working correctly even if `onEnterBackground` isn't called preventing long sessions (it assumes you're in the background).

Not sure I understand how the threading changes will make a difference. Can you please explain more. Thanks

Reply:
> Event before your changes the call to startNewSessionIfNeeded(timestamp) should be working correctly even if onEnterBackground isn't called preventing long sessions (it assumes you're in the background).

Yes, you are right.

Before my changes data races were possible (`inForeground` was set in log thread and can be read in main/log thread). Now all writes/reads are moved to main thread.